src/docs/deployment/macos.md

```diff
@@ -206,6 +207,217 @@ on TestFlight, or go ahead and release your app to the App Store.
+and [Codemagic CLI Tools][codemagic_cli_tools] executed in a terminal
+in the Flutter project directory
+
+<ol markdown="1">
+<li markdown="1">
+
+Install the Codemagic CLI tools:
+```bash
+pip3 install codemagic-cli-tools
+```
+
+</li>
+<li markdown="1">
+
+You'll need to generate an [App Store Connect API Key][appstoreconnect_api_key]
+with App Manager access to automate operations with App Store Connect. To make
+subsequent commands more concise, set the following environment variables from
+the new key: issuer id, key id, and API key file.
+
+```bash
+export APP_STORE_CONNECT_ISSUER_ID=aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee
+export APP_STORE_CONNECT_KEY_IDENTIFIER=ABC1234567
+export APP_STORE_CONNECT_PRIVATE_KEY=`cat /path/to/api/key/AuthKey_XXXYYYZZZ.p8`
+```
+
+</li>
+<li markdown="1">
+
+You'll need to export or create a Mac App Distribution and a Mac Installer
+Distribution certificate to perform code signing and package a build archive.
+
+If you have existing [certificates][devportal_certificates], you can export the
+private keys by executing the following command for each certificate:
+
+```bash
+openssl pkcs12 -in <certificate_name>.p12 -nodes -nocerts | openssl rsa -out cert_key
+```
+
+Or you can create a new private key by executing
+
+```bash
+ssh-keygen -t rsa -b 2048 -m PEM -f cert_key -q -N ""
+```
+
+and later have CLI tools automatically create a new Mac App Distribution and
+Mac Installer Distribution certificate. You can use the same private key for
+each new certificate.
+
+</li>
+<li markdown="1">
+
+Fetch the code signing files from App Store Connect
+
+```bash
+app-store-connect fetch-signing-files YOUR.APP.BUNDLE_ID \
+    --platform MAC_OS \
+    --type MAC_APP_STORE \
+    --certificate-key=@file:/path/to/cert_key \
+    --create
+```
+
+Where `cert_key` is either your exported Mac App Distribution certificate private key
+or a new private key which will automatically generate a new certificate. 
+
+</li>
+<li markdown="1">
+
+If you do not have a Mac Installer Distribution certificate, you can create a new certificate by executing  
+
+```bash
+app-store-connect create-certificate \
+    --type MAC_INSTALLER_DISTRIBUTION \
+    --certificate-key=@file:/path/to/cert_key \
+    --save
+```
+
+with `cert_key` of the private key you created earlier.
+
+</li>
+<li markdown="1">
+
+Fetch the Mac Installer Distribution certificates:
+
+```bash
+app-store-connect list-certificates \
+    --type MAC_INSTALLER_DISTRIBUTION \
+    --certificate-key=@file:/path/to/cert_key \
+    --save
+```
+
+</li>
+<li markdown="1">
+
+Set up a new temporary keychain to be used for code signing:
+
+```bash
+keychain initialize
+```
+
+{{site.alert.warning}}
+**Restore Login Keychain!**
+After running `keychain intialize` you **must** run
+
+```bash
+keychain use-login
+```
+
+in order set your login keychain as the default to avoid potential authentication issues with apps on your machine.
+{{site.alert.end}}
```

Comment:
```suggestion
{{site.alert.secondary}}
  **Restore Login Keychain!**
  After running `keychain initialize` you **must** run the following:<br>

  `keychain use-login`

  This sets your login keychain as the default to avoid potential
  authentication issues with apps on your machine.
{{site.alert.end}}
```